pinot-core/src/main/java/org/apache/pinot/core/minion/SegmentPurger.java

```diff
@@ -100,14 +105,28 @@ public File purgeSegment()
         config.setSegmentTimeUnit(segmentMetadata.getTimeUnit());
       }
 
+      // Generate inverted index if it exists in the original segment
+      // TODO: once the column metadata correctly reflects whether inverted index exists for the column, use that
```

Comment:
In v3 segments we shiould be able to read the index file to see if inverted index exists for the column?

Reply:
SegmentDirectory is an abstraction layer on top of files, which can handle all segment format properly.